src/backend/config.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2023 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+
+from fastapi.logger import logger
+
+
+def get_s3gw_address() -> str:
+    """Obtain s3gw service address from environment, and validate format."""
+
+    addr = os.environ.get("S3GW_SERVICE_ADDRESS")
+    if addr is None:
+        logger.error("S3GW_SERVICE_ADDRESS env variable not set!")
+        raise Exception()
+    m = re.fullmatch(r"https?://[\w.-]+(?:\.[\w]+)?(?::\d+)?/?", addr)
+    if m is None:
+        logger.error(f"Malformed s3gw address: {addr}")
```

Comment:
I think that's more understandable.
```suggestion
        logger.error(f"Malformed address in S3GW_SERVICE_ADDRESS env variable: {addr}")
```

Reply:
I don't think this adds additional value tbh